src/model/reference.ml

```diff
@@ -128,6 +128,31 @@ let tokenize location s =
         scan_identifier started_at
           (open_parenthesis_count - 1)
           (index - 1) tokens
+    | '"' ->
+        (* Note: Don't worry about the seemingly contradictions about variable names usage particularly
+           in this pattern branch, I'll explain.
+           Ideally, we are calculating the identifier from String operations which are independent on how
+           the parser operates; That is to say the parser parses from right to left (see immediate comment
+           above the tokenize function), where as, for our case of calculating the identifier, we consider
+           left to right. We do this specially to keep the identifier (page name in this case) intact.
+
+           However, the if ... else block operations, we return to then concept on how the parser works,
+           because of the scan_kind and scan_identifier calls that are attached to the parser.
+        *)
+        let f_occurence = String.index s '"' in
+        let l_occurence = String.rindex_from s f_occurence '"' in
+        let off_set = f_occurence + 1 in
+        let length = started_at - off_set - 1 in
+        let identifier = String.trim (String.sub s off_set length) in
+        (* The if ... then conditions specially handles cases like "foo"."bar",and
+           it should be checked first, otherwise, and exception will be thrown*)
+        if s.[0] = '"' && s.[f_occurence] = '"' then
```

Comment:
I’m still failing to handle a case like `{!”Foo”. “Bar”}` whose test is at https://github.com/ocaml/odoc/pull/622/commits/e4a86e69850fb694aa11fffadfae6030ecbd5a7f#diff-f59d447ba8018121335d4cfbebc893d052260724d49e9c187d9f4d7e2f7685ff

Reply:
Done!